src/signet.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <signet.h>
+
+#include <consensus/merkle.h>
+#include <consensus/params.h>
+#include <hash.h>
+#include <policy/policy.h>
+#include <primitives/block.h>
+#include <script/interpreter.h>
+#include <util/system.h>
+
+bool g_signet_blocks = false;
+CScript g_signet_blockscript;
+
+// Signet block solution checker
+bool CheckBlockSolution(const CBlock& block, const Consensus::Params& consensusParams)
+{
+    std::vector<uint8_t> signet_data;
+    if (!block.GetWitnessCommitmentSection(SIGNET_HEADER, signet_data)) {
+        return error("CheckBlockSolution: Errors in block (block solution missing)");
+    }
+    if (!CheckBlockSolution(GetSignetHash(block), signet_data, consensusParams)) {
+        return error("CheckBlockSolution: Errors in block (block solution invalid)");
+    }
+    return true;
+}
+
+bool CheckBlockSolution(const uint256& signet_hash, const std::vector<uint8_t>& signature, const Consensus::Params& params)
+{
+    SimpleSignatureChecker bsc(signet_hash);
+    CScript solution = CScript(signature.begin(), signature.end());
+    return VerifyScript(solution, g_signet_blockscript, nullptr, STANDARD_SCRIPT_VERIFY_FLAGS, bsc);
```

Comment:
I don't think you should use standard flags directly. These can be introduced and removed, at least the policy ones.

Instead use `GetBlockScriptFlags` or pass the flags in to get the currently enforced script flags.

Reply:
Good point!